modules/core/04-channel/keeper/upgrade.go

```diff
@@ -313,6 +313,35 @@ func (k Keeper) ChanUpgradeAck(
 	return nil
 }
 
+// writeUpgradeTimeoutChannel restores the channel state of an initialising chain in the event that the counterparty chain has passed the timeout set in ChanUpgradeInit to the state before the upgrade was proposed.
+// Auxiliary upgrade state is also deleted.
+// An event is emitted for the handshake step.
+func (k Keeper) writeUpgradeTimeoutChannel(
+	ctx sdk.Context,
+	portID, channelID string,
+) error {
+	defer telemetry.IncrCounter(1, "ibc", "channel", "upgrade-timeout")
+
+	channel, found := k.GetChannel(ctx, portID, channelID)
+	if !found {
+		panic(fmt.Sprintf("could not find existing channel when updating channel state in successful ChanUpgradeTimeout step, channelID: %s, portID: %s", channelID, portID))
+	}
+
+	upgrade, found := k.GetUpgrade(ctx, portID, channelID)
+	if !found {
+		panic(fmt.Sprintf("could not find existing upgrade when cancelling channel upgrade, channelID: %s, portID: %s", channelID, portID))
+	}
+
+	if err := k.AbortUpgrade(ctx, portID, channelID, types.NewUpgradeError(channel.UpgradeSequence, types.ErrUpgradeTimeout)); err != nil {
+		return errorsmod.Wrap(types.ErrUpgradeRestoreFailed, fmt.Sprintf("failed to restore channel, channelID: %s, portID: %s", channelID, portID))
```

Comment:
Since `ErrUpgradeRestoreFailed` already contains the text that the restore failed and so that we don't lose the error information returned by `AbortUpgrade`:

```suggestion
		return errorsmod.Wrapf(types.ErrUpgradeRestoreFailed, "err: %v", err)
```